src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -0,0 +1,322 @@
+import org.joda.time.format.ISODateTimeFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class RequestByInstanceIdResource extends Resource {
+  private UserRepository userRepository;
+  private Clients clients;
+  private LocationRepository locationRepository;
+  private LoanRepository loanRepository;
+  private final Logger log;
+
+  public RequestByInstanceIdResource(HttpClient client) {
+    super(client);
+    log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(
+      "/circulation/requests/instances", router);
+
+    routeRegistration.create(this::createInstanceLevelRequests);
+  }
+
+
+  void createInstanceLevelRequests(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+
+    clients = Clients.create(context, client);
+    locationRepository = new LocationRepository(clients);
+    userRepository = new UserRepository(clients);
+    loanRepository = new LoanRepository(clients);
+
+    final Result<RequestByInstanceIdRequest> requestByInstanceIdRequestResult =
+      RequestByInstanceIdRequest.from(routingContext.getBodyAsJson());
+
+    if(requestByInstanceIdRequestResult.failed()) {
+      ResponseWritableResult<Object> failed = Result.failed(requestByInstanceIdRequestResult.cause());
+      failed.writeTo(routingContext.response());
+      return;
+    }
+
+    final RequestByInstanceIdRequest requestByInstanceIdRequest =requestByInstanceIdRequestResult.value();
+    ItemByInstanceIdFinder finder = new ItemByInstanceIdFinder(clients.holdingsStorage(), clients.itemsStorage());
+
+    String pickupServicePointId = requestByInstanceIdRequest.getPickupServicePointId().toString();
+    final CompletableFuture<Result<Collection<Item>>> items = finder.getItemsByInstanceId(requestByInstanceIdRequest.getInstanceId().toString());
+
+    final CompletableFuture<Result<Collection<Item>>> availableItems = items.thenApply(r -> r.next(this::getfilteredAvailableItems));
+
+    availableItems.thenCompose(r -> r.after( collectionResult -> findItemsWithMatchingServicePointId(
+                                                                  pickupServicePointId, collectionResult, locationRepository)))
+      .thenApply( r -> r.next( itemsFound -> instanceToItemRequests(requestByInstanceIdRequest, itemsFound)))
+      .thenCompose( r -> r.after( requests -> placeRequests(requests, clients)))
+      .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
+      .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
+      .thenApply(CreatedJsonResponseResult::from)
+      .thenAccept(result -> result.writeTo(routingContext.response()))
+      .exceptionally( err -> {
```

Comment:
Under what circumstances are exceptions expected to fall through to here?

Unless they relate to writing the HTTP response, could we send them on to the client?

Reply:
I was trying to be safe and add a catch-all handler and send back a harmless message to the client.

Reply:
What confuses me slightly about that, is the `exceptionally` block does not respond to the client

Reply:
I'll remove it, then, if it doesn't serve any good purpose.

Reply:
I think it is a good call to add it.

I have been considering doing it on every top level chain like this.

I think it is more that it needs to write a standard server error response, as well as log